k8s/demo/percona/tutorial-running-percona-pod-on-openebs.md

```diff
@@ -1 +1,193 @@
+
+``` 
+```
+karthik@MayaMaster:~$ kubectl get pod
+NAME                                   READY     STATUS    RESTARTS   AGE
+maya-apiserver-1633167387-5ss2w        1/1       Running   0          24s
+openebs-provisioner-1174174075-f2ss6   1/1       Running   0          23s
+
+```
+Check whether the storage classes are applied successfully
+
+```
+karthik@MayaMaster:~$ kubectl get sc
+NAME              TYPE
+openebs-basic     openebs.io/provisioner-iscsi
+openebs-jupyter   openebs.io/provisioner-iscsi
+openebs-percona   openebs.io/provisioner-iscsi
+```
+
+## Step-2: Run percona-mysql pod with openebs storage
+
+Use OpenEBS as persistent storage for the percona pod by selecting an OpenEBS storage class in the persistent volume claim. A sample percona pod yaml (with container attributes and pvc details) is available in the openebs git repo (cloned in the previous steps)
+
+Apply the percona pod yaml 
+
+```
+cd demo/percona
+kubectl apply -f demo-percona-mysql-pvc.yaml
+```
+Verify that the openebs storage pods, i.e., the jiva controller and jiva replicas are created and the percona pod is running succesfully
+
+```
+karthik@MayaMaster:~$ kubectl get pods
+NAME                                                             READY     STATUS    RESTARTS   AGE
+maya-apiserver-1633167387-5ss2w                                  1/1       Running   0          4m
+openebs-provisioner-1174174075-f2ss6                             1/1       Running   0          4m
+percona                                                          1/1       Running   0          2m
+pvc-016e9a68-71c1-11e7-9fea-000c298ff5fc-ctrl-2825810277-rjmxh   1/1       Running   0          2m
+pvc-016e9a68-71c1-11e7-9fea-000c298ff5fc-rep-2644468602-92lfg    1/1       Running   0          2m
+pvc-016e9a68-71c1-11e7-9fea-000c298ff5fc-rep-2644468602-rm8mz    1/1       Running   0          2m
+```
+
+Note: It may take some time for pods to start as the images need to be pulled and instantiated. This is also dependent on the network speed
+
+## Step-3: Run a database client container to generate SQL load 
+
+To test the pod, you can run a kubernetes job, in which a mysql client container runs a load generation script (which in turn performs simple sql queries) to simulate storage traffic. 
+
+Follow the below sequence of steps to achieve this. It can be run on any node _in_ the kubernetes cluster. 
+
+Get the IP address of the percona application pod. This can be obtained by executing kubectl describe on the percona pod
+
+```
+karthik@MayaMaster:~$ kubectl describe pod percona | grep IP
+IP:             10.44.0.3
+``` 
+
+Edit the below line in sql-loadgen job yaml to pass the desired load duration and percona pod IP as arguments.
+In this example, the job performs sql queries on pod with IP address 10.44.0.3 for 300s
+
+```
+args: ["-c", "timelimit -t 300 sh MySQLLoadGenerate.sh 10.44.0.3 > /dev/null 2>&1; exit 0"]
```

Comment:
Yes Amit - we have done as part of our automated testcase which runs in the CI. Ref: https://github.com/openebs/openebs/blob/master/e2e/ansible/playbooks/hyperconverged/test-k8s-percona-mysql-pod/k8s-percona-pod.yml#L82

However, this being a tutorial explaining the workflow of deploying application pod and running load - we chose to make it an explicit step